bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/IStructuredSelection.java

```diff
@@ -57,4 +60,15 @@ public interface IStructuredSelection extends ISelection, Iterable {
 	 * @return the selected elements as a list
 	 */
 	public List toList();
+
+	/**
+	 * Returns the elements in this selection as a <code>Stream</code>.
+	 *
+	 * @return the selected elements as a stream
+	 * @since 3.31
```

Comment:
Since this is an API addition, don't we need a minor version bump here?

Reply:
> Since this is an API addition, don't we need a minor version bump here?

API tools says no and complains otherwise, so who m I to fight the API tools? :-D

Reply:
A minor bump is indeed necessary.
There is a new .api_filter added for this method. If you remove the API filter, don't APT Tools warn you about it?

Reply:
The API filter is because otherwise it wants a **major** (4.0.0) version bump. If I increment the version anyways it complains that the bump is unnecessary and i need another filter for that because it complains that version already was bumped before....

Reply:
Make IStructuredSelection generic?  Noooo!  😱   Mostly one gets a structured selection from/in places where there is no known T...

Reply:
And how does it matter if you don't know `T`  in contrast of having no `T` possibly defined? ;-)

Beside that this might be true for some platform usages, but the usual usage is you create a tree and add a selection listener to it, so if the Tree would be parameterized with `<T>` and my Selection Listener would as well be parametrized with `<T>` my selection can easily be parametrized to the known type `T` ... nerveless this is absolut **NOT IN SCOPE** of this change...!